components/user_profile/user_profile.tsx

```diff
@@ -61,13 +69,15 @@ export default class UserProfile extends PureComponent {
             userId,
         } = this.props;
 
-        let name;
-        if (displayUsername) {
+        let name: React.ReactNode;
+        if (user && displayUsername) {
             name = `@${(user.username)}`;
         } else {
             name = overwriteName || displayName || '...';
         }
 
+        const ariaName: string = typeof name === 'string' ? name.toLowerCase() : '';
```

Comment:
Same question here. I want to make sure we're not removing the `aria-label` in a case where we had one before.

Reply:
Yes, this is also an interesting one. I suspect this might have been another subtle bug that the type checker is helping us to identify.

The issue here is that `overwriteName` might be a `React.node` or a `string`. When it's a `React.node` also `name` will be a `React.node` and we can't really do `toLowerCase()` on it.

Here we might consider to explicitly expose an `ariaName` prop at the component level and enforce that to be a string. This way, even when a node is passed to `overwriteName` we can still control the aria label.

What do you think?

Reply:
I'm looking into this a bit further, and it looks like we don't run into a case where `displayPopover` is true and `name` is not a string. Both this and the above case only get weird when we're using this component for System posts. Technically, the `overwriteName` isn't even used in that place, but we're passing a `FormattedMessage` for it anyway. In any other case, `overwriteName` is a string if it's provided.

I think we can just blame this on the component being old. We'll be re-writing it at some point as part of a larger effort to make better reusable components, so I think we can just address this then. As it is, this'll behave the same as before, although it won't cause a runtime error if it's used incorrectly later on.